rfcs/field-selection.md

```diff
@@ -0,0 +1,1453 @@
+type.
+
+### Multiple Field Reference - Fieldname Matches
+In this scenario, the input fields `UserInput.firstName` and `UserInput.lastName` are semantically
+equivalent to the fields `User.firstName` and `User.lastName`.
+
+```graphql
+type Query {
+    findUserByName(user: UserInput!): User @lookup
+}
+
+input UserInput {
+    firstName: String!
+    lastName: String!
+}
+
+type User {
+    firstName: String!
+    lastName: String!
+}
+```
+
+### Multiple Field Reference - Fieldname is Different
+Here, the input fields `UserInput.firstName` and `UserInput.lastName` are semantically equivalent to
+the fields `User.givenName` and `User.familyName`.
+
+```graphql
+type Query {
+    findUserByName(user: UserInput!): User @lookup
+}
+
+input UserInput {
+    firstName: String!
+    lastName: String!
+}
+
+type User {
+    givenName: String!
+    familyName: String!
+}
+```
+
+### Multiple Field Reference - Output field is Deeper in the Tree
+This scenario involves the input fields `UserInput.firstName` and `UserInput.lastName` being
+semantically equivalent to the fields `Profile.firstName` and `User.lastName`.
+
+```graphql
+type Query {
+    findUserByName(user: UserInput!): User @lookup
+}
+
+input UserInput {
+    firstName: String!
+    lastName: String!
+}
+
+type User {
+    profile: Profile
+    lastName: String!
+}
+
+type Profile {
+    firstName: String!
+}
+```
+
+### Multiple Field Reference - Output field is Deeper in the Tree with Abstract Type
+In this case, the input fields `UserInput.firstName` and `UserInput.lastName` correspond to either
+`EntraProfile.firstName` or `AdfsProfile.firstName`, and `User.lastName`.
+
+```graphql
+type Query {
+    findUserByName(user: UserInput!): User @lookup
+}
+input UserInput {
+    firstName: String!
+    lastName: String!
+}
+
+type User {
+    profile: Profile
+    lastName: String!
+}
+
+union Profile = EntraProfile | AdfsProfile
+
+type EntraProfile {
+    firstName: String!
+}
+
+type AdfsProfile {
+    name: String!
+}
+```
+
+### Multiple Field Reference - Output field is Deeper in the Tree with Abstract Type and Fieldname is Different
+Here, the input fields `UserInput.firstName` and `UserInput.lastName` correspond to either
+`EntraProfile.name` or `AdfsProfile.userName`, and `User.lastName`.
+
+```graphql
+type Query {
+    findUserByName(user: UserInput!): User @lookup
+}
+
+input UserInput {
+    firstName: String!
+    lastName: String!
+}
+
+type User {
+    profile: Profile
+    lastName: String!
+}
+
+union Profile = EntraProfile | AdfsProfile
+
+type EntraProfile {
+    name: String!
+}
+
+type AdfsProfile {
+    userName: String!
+}
+```
+
+### Multiple Field Reference - Input field is Deeper in the Tree
+This introduces additional complexity as the input object requires reshaping. The input fields
+`UserInput.info.firstName` and `UserInput.name` correspond to the fields `User.firstName` and
+`User.name`.
```

Comment:
Alternatively you could use an operation expression such as `>findUserByName(user>info>firstName:)`, e.g.

> This introduces additional complexity as the input object requires reshaping. The input fields `>findUserByName(user>info>firstName:)` and `>findUserByName(user>name:)` correspond to the fields `User.firstName` and `User.name`.